packages/core/src/persist.ts

```diff
@@ -1,20 +1,55 @@
-import { join, dirname } from "node:path";
+import { resolve, join, dirname } from "node:path";
 import { mkdirSync, renameSync, existsSync, copyFileSync, readdirSync } from "node:fs";
 import Logger from "./util/logger";
+import * as darwin from "./darwin";
 
 const logger = new Logger("core/persist");
 
-export default function persist(asarPath: string) {
+export default async function persist(asarPath: string) {
   try {
-    if (process.platform === "win32") {
-      persistWin32(asarPath);
+    persistAsar(asarPath);
+
+    if (process.platform === "darwin") {
+      // the asar is at Discord.app/Contents/Resources/app.asar
+      const inferredBundlePath = resolve(join(dirname(asarPath), "..", ".."));
+      await postPersistSign(inferredBundlePath);
     }
   } catch (e) {
     logger.error(`Failed to persist moonlight: ${e}`);
   }
 }
 
-function persistWin32(asarPath: string) {
+async function postPersistSign(bundlePath: string) {
+  if (process.platform !== "darwin") {
+    logger.error("Ignoring call to postPersistSign because we're not on Darwin");
+    return;
+  }
+
+  logger.debug("Inferred bundle path:", bundlePath);
+
+  if (await darwin.verify(bundlePath, { verbosityLevel: 3 })) {
+    logger.warn("Bundle is currently passing code signing, no need to sign");
+    return;
+  } else {
+    logger.debug("Bundle no longer passes code signing (this is expected)");
+  }
+
+  await darwin.sign(bundlePath, {
+    deep: true,
+    force: true,
+    // TODO: let this be configurable
+    identity: "Moonlight",
```

Comment:
Minor nitpick: the branding is "moonlight", all lowercase. I should establish some branding guidelines someday...